File: apps/ios/AgentSettings.swift
import Foundation

struct AgentSettings {
    private static let serverKey = "open_voice_agent_server_url"
    private static let apiKeyKey = "open_voice_agent_api_key"
    
    // Default fallback (can be overwritten in App Settings or via 1-Prompt pairing)
    static let defaultServerURL = "http://100.120.75.115:8001"
    static let defaultApiKey = "kathy-voice-secure-token"
    
    static var serverBaseURL: String {
        get {
            UserDefaults.standard.string(forKey: serverKey) ?? defaultServerURL
        }
        set {
            UserDefaults.standard.set(newValue, forKey: serverKey)
        }
    }
    
    static var apiKey: String {
        get {
            UserDefaults.standard.string(forKey: apiKeyKey) ?? defaultApiKey
        }
        set {
            UserDefaults.standard.set(newValue, forKey: apiKeyKey)
        }
    }
    
    static var locationHttpURL: URL {
        let base = serverBaseURL.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
        return URL(string: "\(base)/api/v1/location/update") ?? URL(string: "\(defaultServerURL)/api/v1/location/update")!
    }
    
    static var voiceWebSocketURL: URL {
        let base = serverBaseURL.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
        let wsScheme = base.hasPrefix("https://") ? base.replacingOccurrences(of: "https://", with: "wss://") : base.replacingOccurrences(of: "http://", with: "ws://")
        return URL(string: "\(wsScheme)/ws/voice") ?? URL(string: "ws://100.120.75.115:8001/ws/voice")!
    }
}

File: apps/android/AgentSettings.kt
package com.openagent.voice

import android.content.Context

object AgentSettings {
    private const val PREFS_NAME = "open_voice_agent_prefs"
    private const val KEY_SERVER_URL = "server_base_url"
    private const val KEY_API_KEY = "api_key"
    
    // Default fallback (can be customized via settings UI or 1-prompt pairing QR)
    const val DEFAULT_SERVER = "http://100.120.75.115:8001"
    const val DEFAULT_API_KEY = "kathy-voice-secure-token"

    fun getServerBaseUrl(context: Context): String {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_SERVER_URL, DEFAULT_SERVER) ?: DEFAULT_SERVER
    }

    fun setServerBaseUrl(context: Context, url: String) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_SERVER_URL, url.trimEnd('/')).apply()
    }

    fun getApiKey(context: Context): String {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_API_KEY, DEFAULT_API_KEY) ?: DEFAULT_API_KEY
    }

    fun setApiKey(context: Context, key: String) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_API_KEY, key.trim()).apply()
    }

    fun getHttpLocationUrl(context: Context): String {
        val base = getServerBaseUrl(context).trimEnd('/')
        return "$base/api/v1/location/update"
    }

    fun getVoiceWebSocketUrl(context: Context): String {
        val base = getServerBaseUrl(context).trimEnd('/')
        val wsScheme = if (base.startsWith("https://")) {
            base.replace("https://", "wss://")
        } else {
            base.replace("http://", "ws://")
        }
        return "$wsScheme/ws/voice"
    }
}
